### PR TITLE
Add route and navbar

### PR DESCRIPTION
**Pull request by ekang0**
### What
- Install react-router-dom
- Add Router, Switch, and Route to App
- Add NavLink for instructions along with basic styling

### Why
- Need to set up routes in app to render components

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "@testing-library/user-event": "^13.5.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-router-dom": "^5.3.4",
```
Comment (aliu1212) [inline package.json:11]: did you handpick this version? It's OK if you did - just wanted to point out that v6 is out and if you have the time for a small challenge I'd use v6 so you're more in tune with the latest and greatest. 

that being said, it's totally optional! It operates differently than v5 so you'd have to invest a little bit of time reading docs on how to use it

File: src/App.js [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
-    <div className="App">
-      <Header />
-      <NavBar />
-      <Instructions />
-    </div>
+    <Router>
+      <div className="App">
+        <Header />
+        <NavBar />
+        <br></br>
```
Comment (aliu1212) [inline src/App.js]: you can achieve the same using `<br />`

File: src/NavBar.js [hunk 3]
```diff
@@ -1,7 +1,32 @@
+import { NavLink } from "react-router-dom";
+
 function NavBar() {
+  const navStyles = {
```
Comment (aliu1212) [inline src/NavBar.js]: normally if you had typescript/eslint on, it would yell at you for having an empty styling block

File: src/NavBar.js [hunk 4]
```diff
@@ -1,7 +1,32 @@
...
+    textDecoration: "none",
+    color: "white",
+    background: "#384b96",
+    display: "inline-block",
+    width: "50%",
+    padding: "20px 0",
+    fontSize: "30px",
+    letterSpacing: "10px",
+    textAlign: "center",
+    lineHeight: "1"
```
Comment (aliu1212) [inline src/NavBar.js]: for a numerical value like this, you don't need to feed it a string

`lineHeight: 1`

File: src/NavBar.js [hunk 5]
```diff
@@ -1,7 +1,32 @@
...
+  };
 
   return (
-    <div>NAVBAR TEST</div>
+    <div className="navBar" style={navStyles}>
```
Comment (aliu1212) [inline src/NavBar.js]: normally having something like `className="navBar"` would achieve the styling, meaning there's something in place to grab styling for a className "navBar". However, we're also feeding it the navStyles object. 

It achieves the goal but optimally, there's a styling library or something along those lines (or straight up css rules) that takes care of loading up the right set of styles for the class. We prefer this b/c we usually only want to reserve the `style` prop to _override_ some style rules, making it very clear "this is a bit different from the norm"

Not something I'd have you take care of right away tbh, but we'll do a CSS crash course type of thing in the near future

**Review by aliu1212 — CHANGES_REQUESTED** [review]
Nice first PR for review! 

How you want to handle this: 

- For comments like the router v5 vs v6, you can reply with a comment. If you decide to stay v5, leave a comment saying so. If you do want to go v6, leave a small comment but also change the package (npm uninstall ..., npm install ...)

- For fixes such as changing `<br></br>` to `<br />`, you can just make the change(s), commit, and push. No need for a reply comment for small and/or straightforward things

- If not sure what to do in response to a comment, reply to the comment with a question

Those are the basics :)

**Comment by aliu1212** [comment]
also, you'd want to make changes and commit/push to **this** feature branch. This PR then automatically gets updated with your changes

File: package.json (hunk 1 above)
Comment (ekang0) [reply package.json:11]: staying with v5

**Comment by aliu1212** [comment]
and when you're done you hit the re-review button next to my name in the upper right :)

File: package.json (hunk 1 above)
Comment (aliu1212) [reply package.json:11]: great! In future scenarios, add a quick explanation, something like

"staying with v5 for now as we don't have the time in this sprint to make the change to v6" 

not a great example but I think you get what I mean (?)

**Review by aliu1212 — APPROVED** [review]

**Comment by aliu1212** [comment]
next steps:
- hit the merge
- delete the branch
- go to your terminal, switch to the main branch
- git pull
- start a new feature branch from your newly updated main

**Pull request merged**